_posts/2023-06-04-not-generating-constructors.md

```diff
@@ -0,0 +1,411 @@
+---
+tags:
+- Subspace
+- C++
+- Performance
+- Memory Safety
+---
+
+# Avoiding Death by a Thousand Constructors
+
+**tl;dr We are going to see how Subspace types allow a zero-cost transition from native pointers to
+bounded view types (which means we can get memory safety) that can't be achieved with the standard
+library types.**
+
+Long ago, C and C++ programmers passed around pointers to arrays, hopefully with a size to indicate
+how large the array was in order avoid reading/writing out of bounds, though frequently not. This is
+still a common pattern, though "modern C++" has provided tools to eliminate this class of memory
+safety bug, though the use of "view" types. These are types like std::string_view or std::span in
+the standard library. Unfortunately, these types don't actually check for using memory out of bounds
+in the C++ standard.

```

Comment:
My thought on reading this is: All that instruction bloat, but without even the safety. Sad. And the argument for not doing the bounds checking is "performance". Sigh.